include/highfive/bits/H5Converter_misc.hpp

```diff
@@ -174,6 +175,34 @@ struct data_converter<
     size_t _dim;
 };
 
+// apply conversion to std::array
+template <typename T, std::size_t S>
+struct data_converter<
+    std::array<T, S>,
+    typename std::enable_if<(
+                         std::is_same<T, typename type_of_array<T>::type>::value)>::type> {
+    inline data_converter(std::array<T,S>& vec, DataSpace& space, size_t dim = 0)
+    : _space(&space), _dim(dim) {
+        assert(_space->getDimensions().size() >= dim);
+        (void)vec;
+    }
+    
+    inline typename type_of_array<T>::type*
+    transform_read(std::array<T, S>& vec) {
+        return &(vec[0]);
+    }
+    
+    inline typename type_of_array<T>::type*
+    transform_write(std::array<T, S>& vec) {
+        return vec.data();
+    }
+    
+    inline void process_result(std::array<T, S>& vec) { (void)vec; }
+    
+    DataSpace* _space;
+    size_t _dim;
```

Comment:
Members not needed, remove.

Reply:
Dropped the `_dim`. I think `_space` is still needed?

Reply:
In the other (when it's used) it appears in process_results, transform_read or transform_write. See the C array converter, it doesn't have anything.

Reply:
Removed.